### PR TITLE
Landing page

### PR DESCRIPTION
**Pull request by wendevlin**
## Proposed change
Add landing page frontend project

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information
- This PR is related to landingpage PR: home-assistant/landingpage#131

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: build-scripts/bundle.cjs [hunk 1]
```diff
@@ -328,4 +328,20 @@ module.exports.config = {
...
+      name: "landing-page" + nameSuffix(latestBuild),
+      entry: {
+        entrypoint: path.resolve(paths.landingPage_dir, "src/entrypoint.js"),
+      },
+      outputPath: outputPath(paths.landingPage_output_root, latestBuild),
+      publicPath: publicPath(latestBuild),
+      isProdBuild,
+      latestBuild,
+      defineOverlay: {
+        __DEMO__: true,
```
Comment (balloob) [inline build-scripts/bundle.cjs:343]: Why is demo set to true?

File: build-scripts/gulp/gather-static.js [hunk 2]
```diff
@@ -194,3 +204,16 @@ gulp.task("copy-static-gallery", async () => {
   copyLocaleData(paths.gallery_output_static);
   copyMdiIcons(paths.gallery_output_static);
 });
+
+gulp.task("copy-static-landing-page", async () => {
+  // Copy app static files
+  fs.copySync(polyPath("public/static"), paths.gallery_output_static);
```
Comment (balloob) [inline build-scripts/gulp/gather-static.js:210]: why are you copying things to gallery output?

File: build-scripts/gulp/gather-static.js [hunk 3]
```diff
@@ -194,3 +204,16 @@ gulp.task("copy-static-gallery", async () => {
...
+  // Copy gallery static files
```
Comment (balloob) [inline build-scripts/gulp/gather-static.js:211]: stale comment?

File: landing-page/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Home Assistant OS Landingpage
+
+On initial startup of Home Assistant, the HAOS needs to download Home Assistant core before the setup can start.
```
Comment (balloob) [inline landing-page/README.md:3]:
```suggestion
On initial startup of Home Assistant, HAOS needs to download Home Assistant core before the setup can start.
```

File: landing-page/src/ha-landing-page.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import { extractSearchParam } from "../../src/common/url/search-params";
+import { onBoardingStyles } from "../../src/onboarding/styles";
+import type { Constructor } from "../../src/types";
+import themesMixin from "../../src/state/themes-mixin";
+import { makeDialogManager } from "../../src/dialogs/make-dialog-manager";
+import { ProvideHassLitMixin } from "../../src/mixins/provide-hass-lit-mixin";
+
+const SCHEDULE_CORE_CHECK_SECONDS = 5;
+
+const ext = <T extends Constructor>(baseClass: T, mixins): T =>
```
Comment (balloob) [inline landing-page/src/ha-landing-page.ts:19]: Why use this helper if you're only passing in 1 entry in `mixins`

File: src/translations/en.json [hunk 6]
```diff
@@ -7161,6 +7161,29 @@
         "next": "Next",
         "finish": "Finish",
         "help": "Help",
+        "prepare": {
```
Comment (balloob) [inline src/translations/en.json:7164]: I'm surprised this is under `page-onboarding`. Shouldn't it be under a `page-preparing` ? It's a standalone app from the onboarding app.

File: build-scripts/gulp/compress.js [hunk 7]
```diff
@@ -60,9 +66,19 @@ const compressHassioZopfli = () =>
...
+  );
 
 gulp.task("compress-app", gulp.parallel(compressAppBrotli, compressAppZopfli));
 gulp.task(
   "compress-hassio",
   gulp.parallel(compressHassioBrotli, compressHassioZopfli)
 );
+gulp.task(
+  "compress-landing-page",
+  gulp.parallel(compressLandingPageBrotli, compressLandingPageZopfli)
```
Comment (bramkragten) [inline build-scripts/gulp/compress.js:83]: Does the go web server we use support this?

Comment (wendevlin) [reply build-scripts/gulp/compress.js:83]: :shrug: Just did the same as hassio build, I will test it out.

Comment (wendevlin) [reply build-scripts/gulp/compress.js:83]: No it doesn't support this, just with 3rd party libs. I will remove it from the build.

File: landing-page/src/landing-page-base-element.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import {
+  getLocalLanguage,
+  getTranslation,
+} from "../../src/util/common-translation";
+import { computeDirectionStyles } from "../../src/common/util/compute_rtl";
+import themesMixin from "../../src/state/themes-mixin";
+import { translationMetadata } from "../../src/resources/translations-metadata";
+import type { HassBaseEl } from "../../src/state/hass-base-mixin";
+
+export class LandingPageBaseElement extends themesMixin(
```
Comment (bramkragten) [inline landing-page/src/landing-page-base-element.ts:20]: Why do we need themes mixin in landing page?

**Comment by bramkragten** [comment]
Can we check if we can share resources between landing page and onboarding?

File: landing-page/src/components/landing-page-logs.ts [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+    return tempLogLine;
+  }
+
+  private async _startLogStream() {
+    this._error = false;
+    this._newLogsIndicator = false;
+    this._ansiToHtmlElement?.clear();
+
+    try {
+      const response = await getSupervisorLogsFollow();
```
[the remark this answers is not in the document]
Comment (wendevlin) [reply landing-page/src/components/landing-page-logs.ts:181]: I had a meeting with @agners that it should be possible and we fallback to observer if it's not available.

File: landing-page/src/data/supervisor.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+export async function getSupervisorLogsFollow() {
+  return fetch("/supervisor/supervisor/logs/follow?lines=500", {
```
[the remark this answers is not in the document]
Comment (wendevlin) [reply landing-page/src/data/supervisor.ts:2]: There is a PR open to allow this: home-assistant/landingpage#129

File: landing-page/src/landing-page-base-element.ts (hunk 8 above)
Comment (wendevlin) [reply landing-page/src/landing-page-base-element.ts:20]: To have HA styles correctly, is there a better way?
Looks like this without: 
image

**Comment by wendevlin** [comment]
> Can we check if we can share resources between landing page and onboarding?

I reuse `onboarding-welcome-links` and `onboarding/styles`

File: landing-page/.eslintrc.json [hunk 11]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../.eslintrc.json"
+}
```
Comment (silamon) [inline landing-page/.eslintrc.json:3]: Remove this file and add a landing-page/eslint.config.mjs file with 
```
import rootConfig from "../eslint.config.mjs";

export default [
  ...rootConfig,
  {
    rules: {},
  },
];
```

**Review by silamon — COMMENTED** [review]
Now that https://github.com/home-assistant/frontend/pull/22221/files is merged:

File: landing-page/src/data/supervisor.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (bramkragten) [reply landing-page/src/data/supervisor.ts:2]: So this PR is now blocked by that PR. You can also remove that functionality from this PR and add it later.

[the remark this answers is not in the document]
Comment (wendevlin) [reply landing-page/src/data/supervisor.ts:2]: :+1: I'll do it seperate

File: landing-page/src/components/landing-page-network.ts [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+      () => this._fetchSupervisorInfo(),
+      SCHEDULE_FETCH_NETWORK_INFO_SECONDS * 1000
+    );
+  }
+
+  private async _fetchSupervisorInfo() {
+    try {
+      const response = await getSupervisorNetworkInfo();
+      if (!response.ok) {
+        throw new Error("Failed to fetch network info");
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-network.ts:99]: We throw an error here, but we never catch or handle it?

File: landing-page/src/components/landing-page-network.ts [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+    });
+    this._scheduleFetchSupervisorInfo();
+  }
+
+  private async _setDns(ev) {
+    const index = ev.target?.index;
+    try {
+      const response = await setSupervisorNetworkDns(index);
+      if (!response.ok) {
+        throw new Error("Failed to set DNS");
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-network.ts:137]: Same, we don't handle this error

File: landing-page/src/components/landing-page-network.ts (hunk 12 above)
Comment (wendevlin) [reply landing-page/src/components/landing-page-network.ts:99]: It is inside a try catch then we set error to true and show an error in the UI.

File: landing-page/src/components/landing-page-network.ts (hunk 13 above)
Comment (wendevlin) [reply landing-page/src/components/landing-page-network.ts:137]: It is inside a try block. But it handles the same as network info error, I can add an additonal error info for failed set dns.

File: landing-page/src/components/landing-page-network.ts (hunk 12 above)
Comment (bramkragten) [reply landing-page/src/components/landing-page-network.ts:99]: I'm not awake yet I think... Keep your try blocks smaller btw, no need to wrap it all in a try.

File: landing-page/src/components/landing-page-network.ts [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+    if (this._getNetworkInfoError) {
+      return html`
+        <ha-alert alert-type="error">
+          <p>${this.localize("network_issue.error_get_network_info")}</p>
+        </ha-alert>
+      `;
+    }
+
+    return html`
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-network.ts:50]: Should we render this if we couldnt fetch the supervisor info?

File: landing-page/src/components/landing-page-network.ts (hunk 13 above)
Comment (bramkragten) [reply landing-page/src/components/landing-page-network.ts:137]: This one is more important I think, as it is an action a user performs, we now show a message cannot get network info when they try to set the dns severs.

File: landing-page/src/components/landing-page-network.ts (hunk 14 above)
Comment (wendevlin) [reply landing-page/src/components/landing-page-network.ts:50]: :thinking: we don't render it if there is `_getNetworkInfoError  = true`

File: landing-page/src/components/landing-page-network.ts [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+        this._networkIssue = false;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      this._getNetworkInfoError = true;
+    }
+
+    fireEvent(this, "value-changed", {
+      value: this._networkIssue || this._getNetworkInfoError,
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-network.ts:127]: Not sure if we should send `_getNetworkInfoError` up to the parent, there was an issue fetching it, but we don't know if there is an issue with the users network perse.

**Comment by wendevlin** [comment]
@bramkragten I readded log streaming, because home-assistant/landingpage#131 was merged yesterday.

File: landing-page/src/components/landing-page-network.ts (hunk 15 above)
Comment (wendevlin) [reply landing-page/src/components/landing-page-network.ts:127]: Yes thats true

File: landing-page/src/components/landing-page-network.ts [hunk 16]
```diff
@@ -140,7 +143,11 @@ class LandingPageNetwork extends LitElement {
     } catch (err) {
       // eslint-disable-next-line no-console
       console.error(err);
-      this._getNetworkInfoError = true;
+      showAlertDialog(this, {
+        title: this.localize("network_issue.failed"),
+        text: this.localize("network_issue.set_dns_failed"),
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-network.ts:148]: Can we use the error message in this dialog to give an idea of why it failed?

File: landing-page/src/components/landing-page-logs.ts [hunk 17]
```diff
@@ -255,9 +265,14 @@ class LandingPageLogs extends LitElement {
     }
   }
 
-  private async _downloadFullLog() {
+  private _downloadLogs() {
     const timeString = new Date().toISOString().replace(/:/g, "-");
-    fileDownload("/observer/logs", `home-assistant_${timeString}.log`);
+    const downloadUrl = this._observerFallback
```
Comment (bramkragten) [inline landing-page/src/components/landing-page-logs.ts:270]: Is there a benefit of not using the observer for downloading the logs?

Comment (wendevlin) [reply landing-page/src/components/landing-page-logs.ts:270]: I'll recheck

Comment (wendevlin) [reply landing-page/src/components/landing-page-logs.ts:270]: Checked, yes download observer logs is even better here, because it downloads all logs.

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**